src/main/java/htsjdk/samtools/SAMSequenceRecord.java

```diff
@@ -139,6 +148,32 @@ private void setSequenceName(final String name) {
     // Private state used only by SAM implementation.
     public void setSequenceIndex(final int value) { mSequenceIndex = value; }
 
+    /** Returns unmodifiable set with alternative sequence names. */
+    @XmlTransient //we use the field instead of getter/setter
+    public Set<String> getAlternativeSequeneNames() {
+        return Collections.unmodifiableSet(mAlternativeSequenceName);
+    }
+
+    /** Adds an alternative sequence name if it is not the same as the sequence name or it is not present already. */
+    public void addAlternativeSequenceName(final String name) {
+        if (!mSequenceName.equals(name) && ! mAlternativeSequenceName.contains(name) ) {
```

Comment:
These should change to store the names in the attributes map so that you don't need special handling for all the places that handle the tags.